api_v3/lib/KalturaResponseProfileCacher.php

```diff
@@ -138,17 +138,32 @@ public static function start(KalturaObject $apiObject, IRelatedObject $object, K
 		$responseProfileKey = $responseProfile->getKey();
 		$key = self::getObjectSpecificCacheKey($object, $responseProfileKey);
 		$responseProfileCacheKey = self::getResponseProfileCacheKey($responseProfileKey, $object->getPartnerId());
-		
-		list($value, $responseProfileCache) = self::get(array($key, $responseProfileCacheKey));
-		
+
+		$objectCache = null;
+		$responseProfileCache = null;
+		$cachedResults = self::get(array($key, $responseProfileCacheKey));
+		if($cachedResults)
+		{
+			$values = array_values($cachedResults);
+			if(sizeof($cachedResults) < 2)
```

Comment:
For a better performance check the most common flow first (in this case it is the positive flow).